tools/circle_plus_gen/lib/json_parser.py

```diff
@@ -12,7 +12,7 @@ def _generate_optimizer(
     options_t_str: str = opt_type.__name__  # e.g. SGDOptionsT
     options_str: str = options_t_str[:-1]  # e.g. SGDOptions
     optimizer_camel: str = options_str.replace("Options", "")  # e.g. SGD
-    optimizer_str: str = re.sub(r'(?<!^)(?=[A-Z])', '_',
+    optimizer_str: str = re.sub(r'(?<=[a-z])(?=[A-Z])', '_',
```

Comment:
* before, re.sub() convert `SGD` to `S_G_D` which is unexpected. 
* now, it didn't change `SGD` as expected.